From 4d11317501231d05d51edaa5f9247404b84e949e Mon Sep 17 00:00:00 2001
From: Marcin Bober <mbober1@gmail.com>
Date: Mon, 25 Nov 2024 19:28:17 +0000
Subject: [PATCH] radxa zero 3E: Enable SPI3-w5500
Upstream-Status: Inappropriate

---
 .../dts/rockchip/rk3566-radxa-zero-3e.dts     | 41 +++++++++++++++++++
 1 file changed, 41 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3e.dts b/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3e.dts
index ffc394e379d6..bbbc4674948b 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3e.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3e.dts
@@ -38,6 +38,47 @@ &uart3 {
 	status = "okay";
 };
 
+&spi3{
+	status = "okay";
+	#address-cells = <1>;
+	#size-cells = <0>;
+	pinctrl-names = "default", "high_speed";
+	pinctrl-0 = <&spi3m1_cs0 &spi3m1_pins>;
+	pinctrl-1 = <&spi3m1_cs0 &spi3m1_pins_hs>;
+	timeouts = <3>, <3>;
+	num-cs = <2>;
+	cs-gpios = <&gpio4 RK_PC6 GPIO_ACTIVE_LOW
+			&gpio4 RK_PD1 GPIO_ACTIVE_LOW>;
+
+	eth_w5500: eth-w5500@0 {
+		status = "okay";
+		compatible = "wiznet,w5500";
+		reg = <0>;
+		spi-max-frequency = <30000000>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&w5500_int_pins>;
+
+		interrupt-parent = <&gpio3>;
+		interrupts = <RK_PB2 IRQ_TYPE_EDGE_FALLING>;
+	};
+
+	spidev@0 {
+		compatible = "rockchip,spidev";
+		status = "okay";
+		reg = <1>;
+		spi-max-frequency = <50000000>;
+	};
+};
+
+&pinctrl {
+	w5500 {
+		w5500_int_pins: w5500-int-pins {
+			rockchip,pins = <3 RK_PB2 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+};
+
 &mdio1 {
 	rgmii_phy1: ethernet-phy@1 {
 		compatible = "ethernet-phy-ieee802.3-c22";
-- 
2.34.1

