From 9048159d456bec424d7050c5de309dc6e46d0a2c Mon Sep 17 00:00:00 2001
From: Marcin Bober <mbober1@gmail.com>
Date: Mon, 28 Oct 2024 19:15:42 +0000
Subject: [PATCH] build dtbo
Upstream-Status: Inappropriate

---
 arch/arm64/boot/dts/rockchip/Makefile | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/Makefile b/arch/arm64/boot/dts/rockchip/Makefile
index e641033d23d1..63d86a0ffb63 100644
--- a/arch/arm64/boot/dts/rockchip/Makefile
+++ b/arch/arm64/boot/dts/rockchip/Makefile
@@ -1,4 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0
+
+DTC_FLAGS := -@
+
 dtb-$(CONFIG_ARCH_ROCKCHIP) += px30-evb.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += px30-engicam-px30-core-ctouch2.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += px30-engicam-px30-core-ctouch2-of10.dtb
-- 
2.34.1

