From c6c2f7bd4db9ec34db1113183ed93ac37428e736 Mon Sep 17 00:00:00 2001
From: Marcin Bober <mbober1@gmail.com>
Date: Sun, 27 Oct 2024 15:31:35 +0000
Subject: [PATCH] disable ethernet
Upstream-Status: Inappropriate

---
 .../dts/rockchip/rk3566-radxa-zero-3e.dts     | 40 -------------------
 1 file changed, 40 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3e.dts b/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3e.dts
index e2740de71f64..ab0dd7f6b57e 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3e.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3e.dts
@@ -14,46 +14,6 @@ aliases {
 	};
 };
 
-&gmac1 {
-	assigned-clocks = <&cru SCLK_GMAC1_RX_TX>, <&cru SCLK_GMAC1>;
-	assigned-clock-parents = <&cru SCLK_GMAC1_RGMII_SPEED>, <&cru CLK_MAC1_2TOP>;
-	clock_in_out = "input";
-	phy-handle = <&rgmii_phy1>;
-	phy-mode = "rgmii";
-	phy-supply = <&vcc_3v3>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&gmac1m1_miim
-		     &gmac1m1_tx_bus2
-		     &gmac1m1_rx_bus2
-		     &gmac1m1_rgmii_clk
-		     &gmac1m1_rgmii_bus
-		     &gmac1m1_clkinout>;
-
-	tx_delay = <0x4f>;
-	rx_delay = <0x26>;
-	status = "okay";
-};
-
-&mdio1 {
-	rgmii_phy1: ethernet-phy@1 {
-		compatible = "ethernet-phy-ieee802.3-c22";
-		reg = <1>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&gmac1_rstn>;
-		reset-assert-us = <20000>;
-		reset-deassert-us = <50000>;
-		reset-gpios = <&gpio3 RK_PC0 GPIO_ACTIVE_LOW>;
-	};
-};
-
-&pinctrl {
-	gmac1 {
-		gmac1_rstn: gmac1-rstn {
-			rockchip,pins = <3 RK_PC0 RK_FUNC_GPIO &pcfg_pull_none>;
-		};
-	};
-};
-
 &spi3 {
 	status = "okay";
 	#address-cells = <1>;
-- 
2.34.1

