From b5d66327742ae65ba93ad2199b7f1ef94d85c6a8 Mon Sep 17 00:00:00 2001
From: Marcin Bober <mbober1@gmail.com>
Date: Wed, 19 Jun 2024 05:11:28 +0000
Subject: [PATCH] add peripherals

Signed-off-by: Marcin Bober <mbober1@gmail.com>
---
 .../dts/rockchip/rk3566-radxa-zero-3e.dts     | 25 +++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3e.dts b/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3e.dts
index e166d66990b9..b56f32e2919a 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3e.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3e.dts
@@ -11,6 +11,15 @@ / {
 	aliases {
 		ethernet0 = &gmac1;
 	};
+
+	w1: onewire {
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_onewire>;
+		compatible = "w1-gpio";
+		gpios = <&gpio1 RK_PA4 GPIO_ACTIVE_HIGH>;
+		status = "okay";
+	};
+
 };
 
 &gmac1 {
@@ -42,10 +51,26 @@ rgmii_phy1: ethernet-phy@1 {
 	};
 };
 
+&i2c3 {
+	status = "okay";
+ clock-frequency = <400000>; /* i2c bus frequency 400 KHz */
+	mpu6050@68 {
+		compatible = "invensense,mpu6050";
+		reg = <0x68>;
+	};
+};
+
 &pinctrl {
 	gmac1 {
 		gmac1_rstn: gmac1-rstn {
 			rockchip,pins = <3 RK_PC0 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
+
+	onewire {
+		pinctrl_onewire: onewire-1 {
+			rockchip,pins = <1 RK_PA4 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+
 };
-- 
2.34.1

